openhands/sdk/conversation/secrets_manager.py

```diff
@@ -0,0 +1,104 @@
+"""Secrets manager for handling sensitive data in conversations."""
+
+import re
+from typing import Callable, Dict
+
+from openhands.sdk.logger import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class SecretsManager:
```

Comment:
This should be a Pydantic BaseModel from the start, so we can serialize it with the conversation